### ISSUE TITLE
How to end a competition? / feedback on first "real" use

### ISSUE THREAD

**Issue by EmFl** [ezBadminton/ezBadmintonAdmin#2]
Hello, me again !
Well we had our first tournament with your software and so first of all thanks ! 
Everything went rather well except I have no idea how to end a competition so I took screenshots to save the score and leaderboard and then cancelled it.
Other questions/remarks:
- I don't know if there is a way to have a 3rd place match in a competition?
- Would you have an interest in:
  - supporting a "gender neutral" category to have singles match with men/women playing against each other (in my club we are not that many so we all play against each other in the tournament) (same for doubles, some teams are mixed, some are not)
  - Being able to change the score required for matches in the middle of the tournament (for example we do match in the groups as 11 points 2 set wins but we often do the semi-finals and up in 21 points). To workaround this I wanted to create a new competition with only the players that made if to the semi finals but I couldn't create a new one with the same category (since it was considered already running). We ended up tracking the score externally and then just inputting a fake score to get the correct winner.

I understand this isn't the "official" way to play but if you're interested in supporting this I'll try and submit a PR if I manage to add this :)

Thank you again

**Comment by Snonky** [comment]
Hi! I'm glad you could utilize the app, I hope everyone had good fun.

A competition ends when the last match has its result entered. A planned feature is a PDF export of results. But that also works in the middle of the tournament. There is no special state that marks a competition as finished.
You will also be able to save/load entire tournaments so you can start a new one and keep the old results.

> I don't know if there is a way to have a 3rd place match in a competition?

Right now that is only possible for the 'Elimination with consolation' mode where you can set how many places to play out. It is planned to give the group tournament (I assume you used this mode) a setting where you can chose the type of elimination tree that follows the group phase. This way you will be able to make a group phase followed by an elimination with consolation and with that a match for 3rd.

> "gender neutral" category

I am not planning on implementing this but the app does not enforce any gender specifications when registering players.

> change the score required for matches in the middle of the tournament

I have thought about that aswell and I like the option. Right now the scoring options are related to the competitons but it could be remodeled to be a relation of the individual matches.

> I wanted to create a new competition with only the players that made if to the semi finals but I couldn't create a new one with the same category

Yeah you'd have to activate the 'Use playing levels' option before starting any competitions. With that you can create arbitrarily many competitions of the same discipline under different categories (or "playing levels").

> I'll try and submit a PR

I'm stoked you want to try to contribute! I am currently in the process of a small refactor to make a multi-client-server model work. I am expecting that I can commit my changes to the public repo in about a week. The best point in time to start making changes would be after that. From there it will be easier to manage collaboration! I'll message you when I get there if you want.

Thanks again for testing the app in the wild :)

**Comment by Snonky** [comment]
@EmFl I just pushed the refactoring changes.

The server is no longer the default PocketBase binary but a modified version. You should be able to just replace your `pocketbase.exe` with the one from the server repository's [releases](https://github.com/ezBadminton/ezBadmintonServer/releases) page.

Feel free to open a new issue for your proposed changes where we can discuss and I can answer questions.
